redisson/src/main/java/org/redisson/RedissonList.java

```diff
@@ -728,14 +719,24 @@ public int addBefore(V elementToFind, V element) {
     public <T> RFuture<Collection<T>> readSortAsync(String byPattern, List<String> getPatterns, SortOrder order) {
         return readSortAsync(byPattern, getPatterns, order, -1, -1);
     }
-    
+
     @Override
     public <T> Collection<T> readSort(String byPattern, List<String> getPatterns, SortOrder order, int offset, int count) {
         return (Collection<T>)get(readSortAsync(byPattern, getPatterns, order, offset, count));
     }
 
+    @Override
+    public <T> Collection<T> readSort(String byPattern, List<String> getPatterns, SortOrder order, int offset, int count, boolean alpha) {
```

Comment:
How will this method look with more flags (..., 2, 3, true, false)?
And what is more, Redisson API will be broken when Redis adds another sorting type.

It isn't best idea to use boolean flag as  a method argument:
https://www.martinfowler.com/bliki/FlagArgument.html
http://www.informit.com/articles/article.aspx?p=1392524

I see two more alternatives: method sufix or enum argument.

Reply:
@hexmind good point. I prefer enum argument. @mrniko what do you say?